pages/searching/csv_export.rst

```diff
@@ -1,17 +1,41 @@
-Clicking on *Downlaod* will create the file based on the specified options and start the download.
+This will open a modal where you can choose the fields the CSV document should contain. The initial fields are based on the message table configured for the search page.
+It is also possible to define a limit. Because messages are loaded in chunks, all chunks up to the one where the limit is reached will be retrieved. Which means the total number of delivered messages can be higher than the defined limit.
 
-.. image:: /images/searching/csv_export_dialog.png
+Clicking on *Start Download* will create the file based on the specified options and start the download.
+
+.. image:: /images/searching/csv_export_dialog_v2.png
+   :align: center
+
+When the search page or saved search contains multiple message tables, you can decide which message table you want to adopt the settings from. In this case you will see the following select when opening the CSV export modal:
+
+.. image:: /images/searching/csv_export_widget_select.png
+   :align: center
+
+Additionally, you can select *Export to CSV* in a message table’s action menu, if you want to adopt its settings directly.
+
+.. image:: /images/searching/csv_export_message_table_action_menu.png
    :align: center
 
-**Hint**: Some Graylog inputs keep the original message in the ``full_message`` field. If you need to export the original message, you
-can do so by clicking on the *List all fields* link at the bottom of the sidebar, and then selecting the ``full_message`` field.
+Exporting search results on a dashboard
+=======================================
+
+The export on a dashboard is slightly different compared to the search page or saved searches. As described in the section :ref:`dashboards-widget-specific-criteria`, widgets on a dashboard have their own search criteria and the search parameters configured in a dashboard’s search bar only function as a filter. Just keep in mind you will always export the search result currently displayed by the message table. If you want to export only the search results the message table’s search criteria is referring to, make sure the dashboard’s filter is not defined.
+
+Because a dashboard filter does not contain all required search parameters, you will always have to select a message table when clicking on *Export to CSV* in a dashboard’s search bar action menu.
+
+Decorator support
+=================
+
+While the CSV export supports fields created by decorators, they are currently not being listed in the fields select options list. When you want to export a decorated field, enter its name in the field select and click on the option *Create field_name*. If you want to be sure a decorated field is available In the context of the current search, you can find a list of all active decorators in a message table’s edit modal.
```

Comment:
"message table’s edit modal" sounds a tad too technical and not everybody might immediately understand what that is. Maybe replace that reference with a description of what somebody would have to do to see the decorators, i.e., "open the edit modal by clicking `Edit` in the message table's context menu and find the decorators in the left sidebar" or something...